diff --git a/var/spack/repos/builtin/packages/curl/package.py b/var/spack/repos/builtin/packages/curl/package.py
index 6c14f344e8..b13bb3d697 100644
--- a/var/spack/repos/builtin/packages/curl/package.py
+++ b/var/spack/repos/builtin/packages/curl/package.py
@@ -127,6 +127,7 @@ def configure_args(self):
         args += self.with_or_without('libssh2')
         args += self.with_or_without('libssh')
         args += self.enable_or_disable('ldap')
+        args += ['--without-ca-bundle', '--without-ca-path']
         return args
 
     def with_or_without_gnutls(self, activated):
