From 2cdc088dba3127fe70c1cb6b21d22f6399e6a68b Mon Sep 17 00:00:00 2001
From: Harmen Stoppels <harmenstoppels@gmail.com>
Date: Tue, 4 May 2021 13:59:30 +0200
Subject: [PATCH] Force certain flags

---
 lib/spack/env/cc | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/lib/spack/env/cc b/lib/spack/env/cc
index 4d8c4644cb..ab2d01b1c7 100755
--- a/lib/spack/env/cc
+++ b/lib/spack/env/cc
@@ -290,6 +290,12 @@ while [ $# -ne 0 ]; do
     fi
 
     case "$1" in
+        -O0|-O|-O1|-O2|-O3|-Os|-Ofast|-g)
+            # drop -O* and -g flags when SPACK_OPTIMIZATION_FLAGS is set
+            if [ -z "$SPACK_OPTIMIZATION_FLAGS" ]; then
+                other_args+=("$1")
+            fi
+            ;;
         -isystem*)
             arg="${1#-isystem}"
             isystem_was_used=true
@@ -404,6 +410,14 @@ while [ $# -ne 0 ]; do
     shift
 done
 
+case "$mode" in
+    cc|ccld)
+        if [ -n "$SPACK_OPTIMIZATION_FLAGS" ]; then
+            other_args+=($SPACK_OPTIMIZATION_FLAGS)
+        fi
+        ;;
+esac
+
 #
 # Add flags from Spack's cppflags, cflags, cxxflags, fcflags, fflags, and
 # ldflags. We stick to the order that gmake puts the flags in by default.
-- 
2.25.1

