diff --git a/lib/spack/env/cc b/lib/spack/env/cc
index 4d8c4644cb..3c4dcb35a7 100755
--- a/lib/spack/env/cc
+++ b/lib/spack/env/cc
@@ -384,6 +384,13 @@ while [ $# -ne 0 ]; do
                 other_args+=("$1")
             fi
             ;;
+        -O0,-O1,-O2,-O3,-Os)
+            # Always optimize for size
+            other_args+=("-Os")
+            ;;
+        -g)
+            # we don't want this.
+            ;;
         *)
             if [[ "$1" = "$dtags_to_strip" ]] ; then
                 :  # We want to remove explicitly this flag
